zwave_js_server/model/node/__init__.py

```diff
@@ -466,8 +468,20 @@ async def async_send_command(
         if wait_for_result or (
             wait_for_result is None and self.status != NodeStatus.ASLEEP
         ):
-            result = await self.client.async_send_command(message, **kwargs)
-            return result
+            result_task = asyncio.create_task(
+                self.client.async_send_command(message, **kwargs)
+            )
+            await asyncio.wait(
+                [result_task, self._status_event.wait()],
+                return_when=asyncio.FIRST_COMPLETED,
+            )
+            try:
+                if self._status_event.is_set() and not result_task.done():
+                    result_task.cancel()
+                    return None
+                return result_task.result()
+            finally:
+                self._status_event.clear()
```

Comment:
What's the reason for clearing here after the call? Is it to allow calls with `wait_for_result` set to True, after a node has already changed status to eg dead?

Reply:
just seems like good hygiene? either we got here because the status event was set or we got here because the response was returned. We clear the event on success so that the net command awaits for the next status change or a response

Reply:
Yeah, but if the event was set no new calls will get here unless they are made with `wait_for_result` set to True, until the event is cleared by the node status.

Reply:
so do you consider this a no-op then?

Reply:
Not at the moment since we have `wait_for_result`.

Reply:
OK I think I get it now. We want the event to be set anytime the node is in the ASLEEP or DEAD status and cleared anytime the node is in the AWAKE or ALIVE status , and we don't want to mess with the event in the command handler  👍🏾